### ISSUE TITLE
Not able to make wazuh agent class works.

### ISSUE THREAD

**Issue by BornTKill** [wazuh/wazuh-puppet#198]
Dear,

When I try to install Wazuh Agent on a puppet node, I get :

Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Unknown variable: '::wazuh::params_agent::default_local_files'. (file: /etc/puppetlabs/code/environments/hids/modules/wazuh/manifests/params_agent.pp, line: 148, column: 44) on node XXXX.

Here my class :

class qs::wazuh {

  class { "wazuh::agent":
    wazuh_register_endpoint => "X.X.X.X",
    wazuh_reporting_endpoint => "X.X.X.X",
  }

  firewall { '100 allow tcp/1515 wazuh agent  registration':
    dport  => [1515],
    proto  => tcp,
    action => accept,
    iniface => 'ens160',
  }

}

Node definition :
node 'XXXX' {
  include qs::wazuh
}

I also tried example from https://documentation.wazuh.com/3.11/deploying-with-puppet/wazuh-puppet-module/index.html#install-agent-via-puppet but it's the same.
I am on lastest stable.
Thank you for you help.

**Comment by rshad** [comment]
Hi @BornTKill!

Could you please try again, because we've just merged the last changes from `master` into `stable`.

I would also appreciate it if you share your puppet version, running in the puppet agent host.

Kr,

Rshad Zhran

**Comment by BornTKill** [comment]
Dear @rshad 

Thank you for your quick answer.
Do I have something to do before retrying ? (Maybe update my Puppefile ?)
This morning I delete my puppet env branch and make a new r10k deploy but result is the same :

root@xxxxx:/etc/puppetlabs/code/environments/hids# dpkg -l | grep puppet
ii  puppet-agent                          6.11.1-1bionic                                  amd64        The Puppet Agent package contains all of the elements needed to run puppet, including ruby, facter, and hiera.
ii  puppet6-release                       6.0.0-5bionic                                   all          Release packages for the Puppet 6 repository
ii  puppetdb                              6.7.3-1bionic                                   all          Puppet Labs puppetdb
ii  puppetdb-termini                      6.7.3-1bionic                                   all          Termini for puppetdb
ii  puppetserver                          6.7.2-1bionic                                   all          Puppet Labs puppetserver

**Comment by jm404** [comment]
Hi @BornTKill,

In your error, the log stats:
```
Error 500 on SERVER: Server Error: Evaluation Error: Unknown variable: '::wazuh::params_agent::default_local_files'. (file: /etc/puppetlabs/code/environments/hids/modules/wazuh/manifests/params_agent.pp, line: 148, column: 44)
```
But the `params_agent.pp` on line 148 shows

https://github.com/wazuh/wazuh-puppet/blob/30ce72896be818c13ea624a38e8905c1dd8e6b9f/manifests/params_agent.pp#L148

So I suspect your local branch was not updated after we merged the `master` branch into the `stable` one, please make sure to update it with a simple `git pull` and let us know if the same error appears.

Sorry for the inconveniences.

Best regards,

Jose

**Comment by BornTKill** [comment]
Dear,

Can you give me Puppetfile configuration I have to use for that  please ?

Sorry for the inconvenience.

**Comment by rshad** [comment]
Hi @BornTKill!

Once you pull the new changes for the branch `stable`, then you can try again with the following simple Puppetfile configuration:

```ruby
node "<node-name>" {
  class { 'wazuh::agent':
      wazuh_register_endpoint => "<manager-ip>",
      wazuh_reporting_endpoint => "<manager-ip>",
      configure_wodle_openscap => false
        }
}
```

Also, it would be useful if you share with use the `DEBUG` mode output when running `puppet agent -t` command:
```
puppet agent -t --debug
```

Kr,

Rshad

**Comment by BornTKill** [comment]
Dear,

I would like but I manage puppet module with Puppetfile.
Can you help me to configure Puppetfile to get the right version ?

mod 'wazuh-wazuh',
  :git => 'https://github.com/wazuh/wazuh-puppet',
  :tag => 'v3.11.1_7.5.1',
  :branch => 'stable'

Is not working.

**Comment by rshad** [comment]
Hi @BornTKill !

Why don't you choose the latest Wazuh available on Puppetlabs? In this case use the `3.11.1` version.

The corresponding Puppetfile would be as follows:

```ruby
forge "https://forgeapi.puppetlabs.com"

mod 'wazuh-wazuh', '3.11.1'
```

Please let me know if it works or not.

Kr,

Rshad

**Comment by rshad** [comment]
Hi @BornTKill,

As we did not receive any reply from your side for more than 2 months, we will close this issue.

Please do not hesitate to open a new issue or to contact us by
- our Slack channel https://wazuh.com/community/join-us-on-slack/ or by 
- our email list https://groups.google.com/forum/#!forum/wazuh 

if you need any further help.

Kind regards,

Rshad
